src/main/java/app/models/Project.java

```diff
@@ -67,18 +67,24 @@ public boolean addStudent(Student student)
         {
             if (!restrictions.contains(student.getProgram()))
             {
-                students.add(student);
-                if(student.getProject() != this){
-                    student.setProject(this);
+                if (!students.contains(student))
+                {
+                    students.add(student);
+                    if(student.getProject() != this){
```

Comment:
does this need to be .equals since it's an object?

Reply:
Fixed